### PR TITLE
EVG-5755 support pulling docker images from registries

### PR DESCRIPTION
**Pull request by john-m-liu**

### CODE REVIEW

File: cmd/parse-host-file/parse-host-file.go [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+		fmt.Println(err.Error())
+		return
+	}
+	if len(hosts) < 1 {
+		fmt.Println("no hosts listed in file")
+		return
+	}
+
+	data := []byte(fmt.Sprintf("docker_host: %s", hosts[0].DNS))
+	ioutil.WriteFile("expansions.yml", data, 0644)
```
Comment (tychoish) [inline cmd/parse-host-file/parse-host-file.go]: what if  this errors?

File: cmd/parse-host-file/parse-host-file.go [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+	}
+	var hosts []host
+	err = json.Unmarshal(bytes, &hosts)
+	if err != nil {
+		fmt.Println(err.Error())
+		return
+	}
+	if len(hosts) < 1 {
+		fmt.Println("no hosts listed in file")
+		return
```
Comment (tychoish) [inline cmd/parse-host-file/parse-host-file.go]: use grip.EmergencyFatal() to get exit codes for all of the early returns.

File: makefile [hunk 3]
```diff
@@ -179,6 +179,13 @@ $(buildDir)/go-test-config:cmd/go-test-config/make-config.go
 	GOPATH=$(shell dirname $(shell pwd)) $(gobin) build -o $@ $<
 #end generated config
 
+# parse a host.create file and set expansions
+parse-host-file:$(buildDir)/parse-host-file
```
Comment (tychoish) [inline makefile]: "expansions.yml":$(buildDir)/parse-host-file

with an alias target if you want.

File: model/distro/distro.go [hunk 4]
```diff
@@ -42,6 +42,18 @@ type Expansion struct {
 	Value string `bson:"value,omitempty" json:"value,omitempty"`
 }
 
+type ContainerImageSettings struct {
+	URL              string `bson:"url"`
+	Method           string `bson:"method"`
+	RegistryUser     string `bson:"registry_user,omitempty"`
+	RegistryPassword string `bson:"registry_password,omitempty"`
+}
```
Comment (tychoish) [inline model/distro/distro.go]: maybe this should be in the cloud package?

File: cmd/parse-host-file/parse-host-file.go (hunk 2 above)
Comment (tychoish) [reply cmd/parse-host-file/parse-host-file.go]: I think without calling something to exit with a non-zero code, you'll endup always exiting 0, which doesn't make sense.

File: makefile (hunk 3 above)
Comment (tychoish) [reply makefile]: I don't see this change.

Comment (john-m-liu) [reply makefile]: It's on line 187

File: makefile [hunk 5]
```diff
@@ -179,6 +179,14 @@ $(buildDir)/go-test-config:cmd/go-test-config/make-config.go
...
+	./$(buildDir)/parse-host-file --file $(HOST_FILE)
+$(buildDir)/parse-host-file:cmd/parse-host-file/parse-host-file.go
+	$(gobin) build -o $@ $<
+"expansions.yml":$(buildDir)/parse-host-file
```
Comment (tychoish) [inline makefile]: nit: remove the quotes, and maybe put it in the build directory?

File: cmd/parse-host-file/parse-host-file.go (hunk 2 above)
Comment (john-m-liu) [reply cmd/parse-host-file/parse-host-file.go]: added explicit exits for error conditions

File: makefile (hunk 5 above)
Comment (john-m-liu) [reply makefile]: done

**Pull request merged**